pkg/ext/store.go

```diff
@@ -2,92 +2,260 @@ package ext
+		res = append(res, cell)
+	}
+	return res
+}
+
+// CreateOrUpdate helps implement [rest.Updater] by handling most of the logic.
 //
-//	apierrors.NewNotFound(ctx.GroupVersionResource.GroupResource(), name)
+// createValidation is used to do some validation on the object before creating
+// it in the store. For example, it will do an authorization check for "create"
+// verb if the object needs to be created.
+// See here for details: https://github.com/kubernetes/apiserver/blob/70ed6fdbea9eb37bd1d7558e90c20cfe888955e8/pkg/endpoints/handlers/update.go#L190-L201
+// Another example is running mutating/validating webhooks, though we're not using these yet.
 //
-// Stores should make use of the various metav1.*Options as best as possible.
-// Those options are the same options coming from client-go or kubectl, generally
-// meant to control the behavior of the stores. Note: We currently don't have
-// field-manager enabled.
-type Store[T runtime.Object, TList runtime.Object] interface {
-	// Create should store the resource to some backing storage.
-	//
-	// It can apply modifications as necessary before storing it. It must
-	// return a resource of the type of the store, but can
-	// create/update/delete arbitrary objects in Kubernetes without
-	// returning them to the user.
-	//
-	// It is called either when a request creates a resource, or when a
-	// request updates a resource that doesn't exist.
-	Create(ctx Context, obj T, opts *metav1.CreateOptions) (T, error)
-	// Update should overwrite a resource that is present in the backing storage.
-	//
-	// It can apply modifications as necessary before storing it. It must
-	// return a resource of the type of the store, but can
-	// create/update/delete arbitrary objects in Kubernetes without
-	// returning them to the user.
-	//
-	// It is called when a request updates a resource (eg: through a patch or update request)
-	Update(ctx Context, obj T, opts *metav1.UpdateOptions) (T, error)
-	// Get retrieves the resource with the given name from the backing storage.
-	//
-	// Get is called for the following requests:
-	// - get requests: The object must be returned.
-	// - update requests: The object is needed to apply a JSON patch and to make some validation on the change.
-	// - delete requests: The object is needed to make some validation on it.
-	Get(ctx Context, name string, opts *metav1.GetOptions) (T, error)
-	// List retrieves all resources matching the given ListOptions from the backing storage.
-	List(ctx Context, opts *metav1.ListOptions) (TList, error)
-	// Watch sends change events to a returned channel.
-	//
-	// The store is responsible for closing the channel.
-	Watch(ctx Context, opts *metav1.ListOptions) (<-chan WatchEvent[T], error)
-	// Delete deletes the resource of the given name from the backing storage.
-	Delete(ctx Context, name string, opts *metav1.DeleteOptions) error
+// updateValidation is used to do some validation on the object before updating it in the store.
```

Comment:
```suggestion
// updateValidation is used to do validation on the object before updating it in the store.
```

Reply:
Reworded those comments so you can have a look at them again.